publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -375,11 +378,28 @@ public void updateResource(Resource resource) {
     // update this method according to current needs.
     public Entity migrate(Entity dataEntry) {
         if (isResourceWithFiles(dataEntry)) {
+            persistLogEntriesIfNeeded((Resource) dataEntry);
             return migrateResourceWithFiles((Resource) dataEntry);
         }
         return dataEntry;
     }
 
+    private void persistLogEntriesIfNeeded(Resource resource) {
+        var userInstance = UserInstance.fromPublication(resource.toPublication());
+        var logEntries = resource.fetchLogEntries(this);
+        if ("nve@5948.0.0.0".equals(resource.getResourceOwner().getUser().toString())) {
+            resource.setResourceEvent(ImportedResourceEvent.fromImportSource(ImportSource.fromBrageArchive("NVE"),
+                                                                   resource.getCreatedDate()));
```

Comment:
I think this is the only way to differentiate between imported publications from NVE in production.